### PR TITLE
EASY-1125 - userId validation

### PR DESCRIPTION
**Pull request by rvanheest**
fixes [EASY-1125](https://drivenbydata.atlassian.net/browse/EASY-1125)

#### When applied it will
* validate that the user is registered in LDAP and therefore is allowed to ingest the current deposit.

#### Where should the reviewer @DANS-KNAW/easy start?
`EasyIngestFlow`

#### How should this be manually tested?
* Deploy everything on deasy
* Submit a deposit with an invalid userId

### CODE REVIEW

File: src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala [hunk 1]
```diff
@@ -85,6 +89,25 @@ object EasyIngestFlow {
     log.info("No viruses found")
   }
 
+  def assertUserIsAllowedToDeposit(implicit s: Settings): Try[Unit] = {
+    val userId = s.ownerId
+    val ldapContext = s.ldapContext
+
+    val searchName = "dc=dans,dc=knaw,dc=nl"
```
Comment (lindareijnhoudt) [inline src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala]: shouldn't this be part of the config?

File: src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala [hunk 2]
```diff
@@ -85,6 +89,25 @@ object EasyIngestFlow {
...
+    val searchFilter = s"(&(objectClass=easyUser)(uid=$userId))"
+    val searchControls = {
+      val sc = new SearchControls()
+      sc.setSearchScope(SearchControls.SUBTREE_SCOPE)
+      sc
+    }
+
+    log.info(s"validating that user '$userId' is allowed to submit the deposit")
```
Comment (lindareijnhoudt) [inline src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala]: just wondering about words, does one submit a deposit? or a dataset? or does one 'deposit a dataset'? compare with r74 in package.scala

File: src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala (hunk 1 above)
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala]: maybe we should. I copied it right from the license-creator and there we have it hardcoded as well. LDAP is a complicated thing though. If you look in the config (= dtap) you find all kinds of `String`s with easy-sword2 and easy-license-creator. Maybe we should get that sorted first? Not sure if I want to invest much time into this right now, though...

File: src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala (hunk 2 above)
Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/ingest_flow/EasyIngestFlow.scala]: Have been thinking about the wording this morning. But not long enough to at least get them to be consistent! It's an interesting choice: as far as I am aware of, from our perspective the depositor deposits a deposit, which is turned into a dataset in our archive. From the user's perspective he submits or deposits a dataset...

Mmm, I'm confused with these terms now! Let me know what you prefer here.

**Pull request merged**